### PR TITLE
Add announcements back end for limited audiences

### PR DESCRIPTION
**Pull request by jlueck-wss**
Added backend functionality for announcements back end.

### CODE REVIEW

File: plugins/announcements-backend/db/migrations/20240815215_announcements_schemas..js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// @ts-check
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.up = async function up(knex) {
+  return knex.schema.table('announcements', table => {
+    table.boolean('isLimitedAudience').nullable();
```
[the remark this answers is not in the document]
Comment (jlueck-wss) [reply plugins/announcements-backend/db/migrations/20240815215_announcements_schemas..js]: Updated the field to be nullable

File: plugins/announcements-backend/src/service/router.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ interface AnnouncementRequest {
   title: string;
   excerpt: string;
   body: string;
+  isLimitedAudiance: boolean;
```
[the remark this answers is not in the document]
Comment (jlueck-wss) [reply plugins/announcements-backend/src/service/router.ts]: Typo is fixed

File: plugins/announcements-backend/src/service/persistence/AnnouncementsDatabase.test.ts [hunk 3]
```diff
@@ -142,6 +142,7 @@ describe('AnnouncementsDatabase', () => {
       excerpt: 'excerpt',
       body: 'body',
       created_at: DateTime.fromISO('2023-10-26T15:28:08.539Z'),
+      isLimitedAudience: true,
```
Comment (acutlip-wss) [inline plugins/announcements-backend/src/service/persistence/AnnouncementsDatabase.test.ts:145]: I imagine the default audience is "not limited" and therefore the default value here should be `false`.

Comment (jlueck-wss) [reply plugins/announcements-backend/src/service/persistence/AnnouncementsDatabase.test.ts:145]: I can change it but it shouldn't matter. The intent of the test is to test if we are able to delete an announcement that is already in the database.

Comment (acutlip-wss) [reply plugins/announcements-backend/src/service/persistence/AnnouncementsDatabase.test.ts:145]: My mistake. I was moving too fast.

File: plugins/announcements-backend/db/migrations/20240815215_announcements_schemas..js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// @ts-check
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.up = async function up(knex) {
+  return knex.schema.table('announcements', table => {
+    table.boolean('isLimitedAudience').defaultTo(false);
```
Comment (acutlip-wss) [inline plugins/announcements-backend/db/migrations/20240815215_announcements_schemas..js:8]: While we can now create announcements again I still think we should avoid having nullable columns when we can avoid them. What do you think about setting the value to false on insert?

Comment (jlueck-wss) [reply plugins/announcements-backend/db/migrations/20240815215_announcements_schemas..js]: Updated to not allow it to not be null and still have a default value of false.

**Review by acutlip-wss — APPROVED** [review]

**Pull request merged**